PlayCover/Views/Settings/IPASourceSettings.swift

```diff
@@ -8,26 +8,29 @@
 import SwiftUI
 
 struct SourceData: Identifiable, Hashable {
-    var id = UUID()
+    var id: UUID
     var source: String
-    var status: SourceValidation = .valid
+    var status: SourceValidation
+    var isEnabled: Bool
 
-    enum SourceDataKeys: String, CodingKey {
-        case source
-    }
 }
 
-extension SourceData: Encodable {
-    func encode(to encoder: Encoder) throws {
-        var container = encoder.container(keyedBy: SourceDataKeys.self)
-        try container.encode(source, forKey: .source)
+extension SourceData: Codable {
+    init(source: String, isEnabled: Bool) {
+        self.id = UUID()
+        self.source = source
+        self.status = .checking
+        self.isEnabled = isEnabled
     }
-}
 
-extension SourceData: Decodable {
     init(from decoder: Decoder) throws {
-        let values = try decoder.container(keyedBy: SourceDataKeys.self)
-        source = try values.decode(String.self, forKey: .source)
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.init(
+            id: try container.decode(UUID.self, forKey: .id),
```

Comment:
`id` and `status` should also be `decodeIfPresent` for backwards compatibility. `status` shouldn't be read from the source file, but instead always be set to `.checking` by default.

Reply:
```
init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        self.init(
            id: try container.decodeIfPresent(UUID.self, forKey: .id) ?? UUID(),
            source: try container.decode(String.self, forKey: .source),
            status: .checking,
            isEnabled: try container.decodeIfPresent(Bool.self, forKey: .isEnabled) ?? true
        )
    }
```

Like this?

Reply:
Yes.